crates/oxc_linter/src/rules/eslint/no_nested_ternary.rs

```diff
@@ -0,0 +1,69 @@
+use oxc_ast::ast::Expression;
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_nested_ternary_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Do not nest ternary expressions.").with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNestedTernary;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallows nested ternary expressions to improve code readability and maintainability.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Nested ternary expressions make code harder to read and understand. They can lead to complex, difficult-to-debug logic.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const result = condition1 ? (condition2 ? "a" : "b") : "c";
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// let result;
+    /// if (condition1) {
+    ///   result = condition2 ? "a" : "b";
+    /// } else {
+    ///   result = "c";
+    /// }
+    /// ```
+    NoNestedTernary,
+    style,
+);
+
+impl Rule for NoNestedTernary {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ConditionalExpression(node) = node.kind() {
+            if matches!(node.alternate, Expression::ConditionalExpression(_))
```

Comment:
i think this should use `get_inner_expression` to ignore parentiisized expressions/ ts non-null-expressions / as expressions ect

Reply:
Nice! updated: https://github.com/oxc-project/oxc/pull/8150/commits/106875ab62fd927f6106159b9549f5327cbb7b91